### PR TITLE
Bump Ruby to v2.7.5

### PR DESCRIPTION
**Pull request by vishaltelangre**
### Why?

Because we want to upgrade the Heroku app from heroku-18 to heroku-20. But heroku-20 doesn't support the current Ruby version 2.6.2 as [mentioned here](https://github.com/gumroad/wilfred/pull/35#issuecomment-1077627443).

### What?

Apart from bumping the Ruby version to v2.7.5, other dependent gem versions have been upgraded/locked as necessary.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -60,7 +61,7 @@ GEM
       multipart-post (>= 1.2, < 3)
     faraday_middleware (0.13.1)
       faraday (>= 0.7.4, < 1.0)
-    ffi (1.10.0)
+    ffi (1.11.1)
```
Comment (vishaltelangre) [inline Gemfile.lock:64]: To silence warnings like this

```
/Users/vishal/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/ffi-1.10.0/lib/ffi/pointer.rb:97: warning: rb_safe_level will be removed in Ruby 3.0
/Users/vishal/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/sassc-2.0.1/lib/sassc/functions_handler.rb:33: warning: rb_safe_level will be removed in Ruby 3.0
/Users/vishal/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/sassc-2.0.1/lib/sassc/script/value_conversion/string.rb:9: warning: rb_safe_level will be removed in Ruby 3.0

```

File: Gemfile.lock [hunk 2]
```diff
@@ -87,7 +88,7 @@ GEM
     mail (2.7.1)
       mini_mime (>= 0.1.1)
     method_source (0.9.2)
-    mime-types (3.2.2)
+    mime-types (3.3.1)
```
Comment (vishaltelangre) [inline Gemfile.lock:91]: To fix warnings like this

```
/Users/vishal/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/mime-types-3.2.2/lib/mime/types/logger.rb:30: warning: `_1' is reserved for numbered parameter; consider another name
```

**Comment by vishaltelangre** [comment]
@chrismaximin _a Please review.

**Review by chrismaximin — APPROVED** [review]
@vishaltelangre _a

**Comment by vishaltelangre** [comment]
We're now using the heroku-20 stack!

**Pull request merged**